Demo/Iterator/CMakeLists.txt

```diff
@@ -1,8 +1,13 @@
 project(iterator)
 
 
-add_executable(iterator CircularIntIterator.cpp )
-add_executable(iterator FilterIntegerIterator.cpp )
-add_executable(iterator ReverseArrayIterator.cpp )
+add_executable(iterator1 CircularIntIterator.cpp )
```

Comment:
Suppose you encounter to a executable called `iterator666`. What would you think about that executable ? The normal answer is "Something connected to the iterators, but who and why someone named it like that ?"

Rename your executable in more descriptive manner (e.g. `FilterIntegerIteratorDemo`)